Metadata fix + a simpler file content option

ijc25 opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/linuxkit/linuxkit/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

Fixed a bug in parsing the input JSON and added a less verbose mechanism to populate a data (i.e. not executable) file 

**- How I did it**

If a file key corresponds to a string rather than a map take that to be the desired file contents and write it with the default rw permissions (moduo `umask`). If the key corresponds to a map then the existing scheme (with separate `content` and `perm` keys) is maintained.

**- How to verify it**

Provide e.g. `{ "foo": { "bar": "some contents" } }` and observe that `/var/config/foo/bar` contains `some contents`.

**- Description for the changelog**
Metadata JSON can now contain a simple string which will be written to the file

**- A picture of a cute animal (not mandatory but encouraged)**
![Faith No More, Angel Dust](https://upload.wikimedia.org/wikipedia/en/2/24/Faith_no_more_angel_dust.jpg)

rneugeba commented on pkg/metadata/main.go:
```diff
@@ -167,25 +167,31 @@ func processUserData(data []byte) error {
+					continue
+				}
+				c = fi["content"].(string)
+				if p, err = strconv.ParseUint(fi["perm"].(string), 8, 32); err != nil {
```
does this set `p` if there is an error?

justincormack replied:
not defined what the value is if `err` set.

rneugeba replied:
NM, misread the code slightly.

ijc25 replied:
Yeah, if err is set it'll hit the `continue` and never look at `p` and skip the file (like it did before)

rneugeba commented:
Actually, could you add the new format to https://github.com/linuxkit/linuxkit/blob/master/docs/metadata.md as well?

ijc25 commented:
I added the docs.

I also changed the default perm from `0666` to `0644` having observed that the default `umask` in Alpine (and therefore the metadata container) is `0022`, so `0644` is the actual result. Since the `umask` used in the metadata container isn't exposed to the user my attempts to document the permissions as `0666-umask` were just getting silly and this makes more sense anyway I think.

ijc25 commented:
@riyazdf did you mean something more than the docs entry with your "would be great to note the default perms for string input" comment?
